src/Symfony/Component/Process/Process.php

```diff
@@ -358,6 +358,11 @@ public function start(?callable $callback = null, array $env = []): void
 
         try {
             $process = @proc_open($commandline, $descriptors, $this->processPipes->pipes, $this->cwd, $envPairs, $this->options);
+
+            // Backwards compatibility. To be removed in version 8.0
+            if (!\is_resource($process) && \is_array($commandline)) {
+                $process = @proc_open($this->buildShellCommandline($commandline), $descriptors, $this->processPipes->pipes, $this->cwd, $envPairs, $this->options);
+            }
```

Comment:
Should we trigger a deprecation here like this?

```php
if (\is_resource($process)) {
    trigger_deprecation('symfony/process', '7.1', 'Executing non-existent commands will no longer start the process and instead throw a "%s" exception in 8.0.', ProcessStartFailedException::class);
}
```

Reply:
Or just document it in the UPGRADE-7.1.md?

Reply:
there's nothing to do IMHO, just revert to the previous behavior

Reply:
We lose a great deal of performance for cases where `proc_open()` fails. That’s why I’d prefer to remove this in a future version.

Reply:
> We lose a great deal of performance for cases where proc_open() fails

Sure but do we have a use case where failures are on the hot path?
If not, things are fine as is. If yes, we should make this opt-in.

Reply:
I’m not sure. If someone has that use case they could probably make a pull request then.

I’d still like the behavior better that an invalid command throws an exception (like an invalid cwd already does). This would be the better API design I think. But that’s your decision of course :)

Reply:
Forward plan looks fine to me. We are tightly bound by BC concerns :)

Reply:
If we want to throw again an exception on invalid command in 8.0, we need to provide an opt-in in 7.x (but not 7.1 as new deprecations cannot be introduced in patch versions) with a deprecation when you keep using the old error handling behavior. We cannot just change the behavior in 8.0.